compiler/rustc_next_trait_solver/src/solve/trait_goals.rs

```diff
@@ -1434,11 +1432,8 @@ where
         self.merge_trait_candidates(candidates)
     }
 
-    fn try_stall_coroutine_witness(
-        &mut self,
-        self_ty: I::Ty,
-    ) -> Option<Result<Candidate<I>, NoSolution>> {
-        if let ty::CoroutineWitness(def_id, _) = self_ty.kind() {
+    fn try_stall_coroutine(&mut self, self_ty: I::Ty) -> Option<Result<Candidate<I>, NoSolution>> {
+        if let ty::Coroutine(def_id, _) = self_ty.kind() {
```

Comment:
doesn't this also needs to stall for `CoroutineClosures`?

Reply:
No, we don't actually use the witness types for any coroutine-closure goals. Cloning/auto-traits for coro-closures don't actually rely on the witness.